### PR TITLE
ROCANA-3400: Add HDP profile to Kite release job

### PR DESCRIPTION
**Pull request by joey**

### CODE REVIEW

**Comment by joey** [comment]
Tested with:

```
KITE_RELEASE_VERSION=1.0.0-cdh5.4.0-rocana1.3.0-rc3 KITE_DEVELOPMENT_VERSION=1.0.0-cdh5.4.0-rocana1.3.0-SNAPSHOT ./rocana-release.sh
```

Tests with -rc1 and -rc2 failed, hence the jump to -rc3. The script does some pushes, so you have to make sure you're in a branch if you want to test the release script.

File: kite-data/kite-data-core/pom.xml [hunk 1]
```diff
@@ -271,5 +271,36 @@
         </plugins>
       </build>
     </profile>
+    <profile>
+      <id>hdp2-release</id>
```
Comment (bradcupit) [inline kite-data/kite-data-core/pom.xml]: I'm confused (so I'm probably going to learn something new). This profile seems identical to the one above it except it has a different `id` and doesn't have the `install-hdp-classifier` execution. Do we need both?

File: kite-data/kite-data-core/pom.xml [hunk 2]
```diff
@@ -271,5 +271,36 @@
...
+      <build>
+        <plugins>
+          <plugin>
+            <groupId>org.apache.maven.plugins</groupId>
+            <artifactId>maven-deploy-plugin</artifactId>
+            <version>${vers.maven-deploy-plugin}</version>
+            <executions>
+              <execution>
+                <id>deploy-hdp-classifier</id>
```
Comment (bradcupit) [inline kite-data/kite-data-core/pom.xml]: I'm guessing it's ok to duplicate the execution id since it's in two different profiles.

Comment (bradcupit) [inline kite-data/kite-data-core/pom.xml]: Ah, found this [here](https://maven.apache.org/guides/mini/guide-configuring-plugins.html):

> Note that while execution id's have to be unique among all executions of a single plugin within a POM, they don't have to be unique across an inheritance hierarchy of POMs. Executions of the same id from different POMs are merged. The same applies to executions that are defined by profiles.

File: kite-data/kite-data-core/pom.xml (hunk 1 above)
Comment (bradcupit) [inline kite-data/kite-data-core/pom.xml]: Ah, I see, the difference is the `repositoryId` and `url`.

So next question: can we control that with a property instead? If we can, and we merge the two profiles, that means we would not put releases in the snapshot repository (which sounds good to me), and since this [line](https://github.com/scalingdata/cdk/pull/23/files#diff-1441627d1cf34e12817f363bf0b54effR50) sets both profiles at once and the maven docs linked to in another comment say the executions get merged since they have the same ID, then I'm guessing we aren't deploying to both the snapshot and release repos anyway.

The only thing I think we'd miss out on by merging the profiles is the `hdp2-release` profile would now get `install-hdp-classifier` execution, whereas right now it doesn't have that.

File: kite-data/kite-data-crunch/pom.xml [hunk 3]
```diff
@@ -318,5 +318,36 @@
         </plugins>
       </build>
     </profile>
+    <profile>
+      <id>hdp2-release</id>
+      <build>
+        <plugins>
+          <plugin>
+            <groupId>org.apache.maven.plugins</groupId>
+            <artifactId>maven-deploy-plugin</artifactId>
```
Comment (bradcupit) [inline kite-data/kite-data-crunch/pom.xml]: I'm still confused why we need these. I think it has something to do with Kite shading Guava, as described here: https://github.com/scalingdata/cdk/pull/20#discussion_r40102178

Though I'm sure you'll explain it very well I think this config is unusual enough that others may get confused too. Maybe worth a comment?

File: kite-data/kite-data-spark/pom.xml [hunk 4]
```diff
@@ -318,5 +318,36 @@
         </plugins>
       </build>
     </profile>
+    <profile>
+      <id>hdp2-release</id>
```
Comment (bradcupit) [inline kite-data/kite-data-spark/pom.xml]: We seem to have added this type of config to each child of kite-data except for flume and hbase. I know we don't use hbase, so if we don't use flume then that makes sense why those two were left out, but I thought we also don't use Spark. So why the change to this file?

**Comment by bradcupit** [comment]
Is there any way to add all of the profile definitions to the kite-data pom since it's a parent of so many poms affected in this commit?

File: kite-data/kite-data-core/pom.xml (hunk 1 above)
Comment (joey) [inline kite-data/kite-data-core/pom.xml]: I can look at setting those by properties. We don't need the `install-hdp-classifier` execution of the install plugin during a release, we only do a deploy.

File: kite-data/kite-data-crunch/pom.xml (hunk 3 above)
Comment (joey) [inline kite-data/kite-data-crunch/pom.xml]: I'll add a comment. The issue is that we only want to deploy the classified version of jars during the HDP build, not the primary artifacts. The only way to do that is to disable deploy of main artifacts which is done in the parent POM and then to add this rule to deploy a specific file.

File: kite-data/kite-data-spark/pom.xml (hunk 4 above)
Comment (joey) [inline kite-data/kite-data-spark/pom.xml]: The Flume and HBase modules are specifically not built in our fork, so there was no need to modify their POMs. We don't use Spark, but there was a low cost to keeping it in there.

**Comment by joey** [comment]
Sadly no. They reference a specific file that needs to exist in order for the deploy to work, but that file only exists in modules that get packaged as a jar. So, when the `kite-data` module runs, it would fail when it tried to deploy a non-existent file.

**Comment by joey** [comment]
Thanks for the tip on the property @bradcupit , that simplified things a lot! I tested this with:

```
KITE_RELEASE_VERSION=1.0.0-cdh5.4.0-rocana1.3.0-rc5 KITE_DEVELOPMENT_VERSION=1.0.0-cdh5.4.0-rocana1.3.0-SNAPSHOT ./rocana-release.sh
```

The extra commits caused by the release process will get squashed out before the merge.

**Comment by bradcupit** [comment]
Ooohhh, nice. It turned out great!
+1
